artemis-server/src/main/java/org/apache/activemq/artemis/core/postoffice/impl/BindingsImpl.java

```diff
@@ -53,7 +54,13 @@
 
    private final Map<SimpleString, Integer> routingNamePositions = new ConcurrentHashMap<>();
 
-   private final Map<Long, Binding> bindingsMap = new ConcurrentHashMap<>();
+   private final Map<Long, Binding> bindingsIdMap = new ConcurrentHashMap<>();
+
+   /**
+    * This is the same as bindingsIdMap but indexed on the binding's uniqueName rather than ID. Two maps are
+    * maintained to speed routing, otherwise we'd have to loop through the bindingsIdMap when routing to an FQQN.
+    */
+   private final Map<SimpleString, Binding> bindingsNameMap = new ConcurrentHashMap<>();
```

Comment:
I'm not aware of the context here, but  considering that are used 2 different concurrent maps I suppose that there is no chance that data would be different between them from a reader pov right?
If not I suppose that's fine

Reply:
Do you mean the same?  I'm not sure I'm following you here.

Reply:
Both maps are used for routing messages, but they are used for different use-cases which I can't see any overlap between so I don't think there's a problem.